### PR TITLE
runtime: fix return data checking + change test fixture passing criteria

### PR DESCRIPTION
**Pull request by ravyu-jump**

### CODE REVIEW

**Review by 0x0ece — DISMISSED** [review]

**Comment by ravyu-jump** [comment]
tests fail now 🙃 don't shoot the messenger

**Review by mjain-jump — COMMENTED** [review]
UBSAN false positives result in a non-zero return code when all the tests should be passing without issue, namely with `linux_clang_icelake`. Instead, I've changed the passing check to also ensure that the program actually executed all of the tests and didn't terminate early.

File: contrib/test/run_test_vectors.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ OBJDIR=${OBJDIR:-build/native/gcc}
 
 if [ "$LOG_PATH" == "" ]; then
   LOG_PATH="`mktemp -d`"
-else
+else
```
Comment (0x0ece) [inline contrib/test/run_test_vectors.sh:11]: nit

File: src/flamenco/runtime/tests/fd_exec_instr_test.c [hunk 2]
```diff
@@ -701,12 +702,13 @@ _diff_effects( fd_exec_instr_fixture_diff_t * check ) {
   }
 
   /* Check return data */
-  if (expected->return_data->size != ctx->txn_ctx->return_data.len) {
+  ulong data_sz = expected->return_data ? expected->return_data->size : 0UL; /* support expected->return_data==NULL */
+  if (data_sz != ctx->txn_ctx->return_data.len) {
```
Comment (0x0ece) [inline src/flamenco/runtime/tests/fd_exec_instr_test.c:706]: nit: spaces

File: src/flamenco/runtime/tests/fd_exec_instr_test.c [hunk 3]
```diff
@@ -701,12 +702,13 @@ _diff_effects( fd_exec_instr_fixture_diff_t * check ) {
...
     check->has_diff = 1;
     REPORTV( WARNING, "expected return data size %lu, got %lu",
-             (ulong) expected->return_data->size, ctx->txn_ctx->return_data.len );
+             (ulong) data_sz, ctx->txn_ctx->return_data.len );
   }
-  else if (expected->return_data->size > 0 ) {
+  else if (data_sz > 0 ) {
```
Comment (0x0ece) [inline src/flamenco/runtime/tests/fd_exec_instr_test.c:711]: nit: spaces

**Review by 0x0ece — APPROVED** [review]

**Review by ibhatt-jumptrading — APPROVED** [review]
lgtm

**Review by ibhatt-jumptrading — APPROVED** [review]

**Pull request merged**